nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -797,7 +797,7 @@ in
       };
 
       proxyCache = mkOption {
-        type = types.submodule {
+        type = types.attrsOf (types.submodule ({ ... }: {
```

Comment:
This is a breaking change, right? You need to provide some warning / breaking change notice in release notes and in the module itself, when did we introduce `proxyCache`? Do we have a NixOS release with it?
I would argue you should keep `proxyCache` as is and introduce `proxyCaches` for multiple proxy caches.
Deprecate `proxyCache` then remove it in the next release.

Reply:
This option has been recently introduced. I think that few people have used this feature and you can do without a warning.

Reply:
Unfortunately, a lot of people are on unstable, it just mean we can deprecate it faster.

Reply:
Renamed optioan and add warning change notice.